core/includes/form.inc

```diff
@@ -4830,38 +4847,47 @@ function theme_file($variables) {
  *   $form structure and set via form_builder()).
  * - form-disabled: Only set if the form element is #disabled.
  *
- * In addition to the element itself, the DIV contains a label for the element
- * based on the optional #title_display property, and an optional #description.
- *
- * The optional #title_display property can have these values:
- * - before: The label is output before the element. This is the default.
- *   The label includes the #title and the required marker, if #required.
- * - after: The label is output after the element. For example, this is used
- *   for radio and checkbox #type elements as set in system_element_info().
- *   If the #title is empty but the field is #required, the label will
- *   contain only the required marker.
- * - invisible: Labels are critical for screen readers to enable them to
- *   properly navigate through forms but can be visually distracting. This
- *   property hides the label for everyone except screen readers.
- * - attribute: Set the title attribute on the element to create a tooltip
- *   but output no label element. This is supported only for checkboxes
- *   and radios in form_pre_render_conditional_form_element(). It is used
- *   where a visual label is not needed, such as a table of checkboxes where
- *   the row and column provide the context. The tooltip will include the
- *   title and required marker.
- *
- * If the #title property is not set, then the label and any required marker
- * will not be output, regardless of the #title_display or #required values.
- * This can be useful in cases such as the password_confirm element, which
- * creates children elements that have their own labels and required markers,
- * but the parent element should have neither. Use this carefully because a
- * field without an associated label can cause accessibility challenges.
+ * In addition to the element itself, the DIV contains:
+ * - a label (#title) for the element, which can be positioned relative to the
```

Comment:
The nesting of this help text is hard for me to follow. We spend 5 paragraphs talking about labels, and then jump back to resume a sentence about what the parent DIV contains. Can we revert most of the changes related to `#title_display` (improvements are still okay), but just reword the introductory paragraph from:

```
 * In addition to the element itself, the DIV contains a label for the element
 * based on the optional #title_display property, and an optional #description.
```

To:
```
In addition to the element itself, the DIV contains a label and a description.
The label and description can be positioned by the #title_display and
#description_display properties respectively.
```

And then separately describe the properties:

```
The optional #title_display property can have these values:
```

and

```
The optional #description_display property can have these values:
```